src/arithmetic/rewrite_simplify.cc

```diff
@@ -99,6 +93,12 @@ TryCompare(const Expr& x, int64_t val) {
   if (dbound->max_value <= val) {
```

Comment:
for enhancement , do we need to change dbound->max_value <= val into dbound->max_value == val? similar  existing issue in line 90.

Reply:
I'm not sure this is really an enhancement, current formulation seems more straightforward, e.g. if max_value were an Expr, we could write `can_prove(max_value <= val)` whereas `can_prove(max_value == val)` would be "less complete".